google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ChangeStreamMutation.java

```diff
@@ -0,0 +1,272 @@
+  }
+
+  /** Creates a new instance of a GC mutation. */
+  static ChangeStreamMutation create(
+      @Nonnull ByteString rowKey,
+      @Nonnull Type type,
+      @Nonnull Timestamp commitTimestamp,
+      int tieBreaker) {
+    Preconditions.checkArgument(
+        type == Type.GARBAGE_COLLECTION,
+        "ChangeStreamMutation without source cluster id must be a garbage collection mutation.");
+    return new ChangeStreamMutation(rowKey, type, null, commitTimestamp, tieBreaker);
+  }
+
+  private void readObject(ObjectInputStream input) throws IOException, ClassNotFoundException {
+    input.defaultReadObject();
+
+    @SuppressWarnings("unchecked")
+    ImmutableList<Entry> deserialized = (ImmutableList<Entry>) input.readObject();
+    this.entries = ImmutableList.<Entry>builder().addAll(deserialized);
+  }
+
+  private void writeObject(ObjectOutputStream output) throws IOException {
+    output.defaultWriteObject();
+    output.writeObject(entries.build());
+  }
+
+  @Nonnull
+  public ByteString getRowKey() {
+    return this.rowKey;
+  }
+
+  @Nonnull
+  public Type getType() {
+    return this.type;
+  }
+
+  public String getSourceClusterId() {
+    return this.sourceClusterId;
+  }
+
+  @Nonnull
+  public Timestamp getCommitTimestamp() {
+    return this.commitTimestamp;
+  }
+
+  public int getTieBreaker() {
+    return this.tieBreaker;
+  }
+
+  public ChangeStreamMutation setToken(@Nonnull String token) {
+    this.token = token;
+    return this;
+  }
+
+  public String getToken() {
+    return this.token;
+  }
+
+  public ChangeStreamMutation setLowWatermark(@Nonnull Timestamp lowWatermark) {
+    this.lowWatermark = lowWatermark;
+    return this;
+  }
+
+  public Timestamp getLowWatermark() {
+    return this.lowWatermark;
+  }
+
+  @Nonnull
+  public List<Entry> getEntries() {
+    return this.entries.build();
+  }
+
+  ChangeStreamMutation setCell(
+      @Nonnull String familyName,
+      @Nonnull ByteString qualifier,
+      long timestamp,
+      @Nonnull ByteString value) {
+    this.entries.add(new SetCell(familyName, qualifier, timestamp, value));
+    return this;
+  }
+
+  ChangeStreamMutation deleteCells(
+      @Nonnull String familyName,
+      @Nonnull ByteString qualifier,
+      @Nonnull TimestampRange timestampRange) {
+    this.entries.add(new DeleteCells(familyName, qualifier, timestampRange));
+    return this;
+  }
+
+  ChangeStreamMutation deleteFamily(@Nonnull String familyName) {
+    this.entries.add(new DeleteFamily(familyName));
+    return this;
+  }
+
+  public RowMutation toRowMutation(@Nonnull String tableId) {
+    Preconditions.checkArgument(
+        token != null && lowWatermark != null,
+        "ChangeStreamMutation must have a continuation token and low watermark.");
+    RowMutation rowMutation = RowMutation.create(tableId, rowKey);
+    for (Entry entry : this.entries.build()) {
+      if (entry instanceof DeleteFamily) {
```

Comment:
maybe use a switch(entry) {
  case DeleteFamily 
...
}

instead

Reply:
Discussed offline. Switching on subclasses needs sdk 16. Let's keep it for now.